MapboxMobileEvents/MMEMetricsManager.m

```diff
@@ -0,0 +1,143 @@
+    self = [super init];
+    if (self) {
+        _metrics = [[MMEMetrics alloc] init];
+    }
+    return self;
+}
+
+- (void)metricsFromEventQueue:(NSArray *)eventQueue {
+    if (self.metrics.dateUTC == nil) {
+        [self updateDateUTC];
+    }
+    
+    if (eventQueue.count > 0) {
+        if (self.metrics.eventCountPerType == nil) {
+            self.metrics.eventCountPerType = [[NSMutableDictionary alloc] init];
+        }
+        
+        self.metrics.eventCountTotal += (int)eventQueue.count;
+        
+        for (MMEEvent *event in eventQueue) {
+            NSNumber *eventCount = [self.metrics.eventCountPerType objectForKey:event.name];
+            eventCount = [NSNumber numberWithInteger:[eventCount integerValue] + 1];
+            [self.metrics.eventCountPerType setObject:eventCount forKey:event.name];
+        }
+    }
+}
+
+- (void)metricsFromEvents:(nullable NSArray *)events error:(nullable NSError *)error {
+    if (error == nil) {
+        self.metrics.requests++;
+    } else {
+        if (events) {
+            self.metrics.eventCountFailed += (int)events.count;
+        }
+        
+        if ([error.userInfo objectForKey:MMEResponseKey]) {
+            NSHTTPURLResponse *response = (NSHTTPURLResponse *)[error.userInfo objectForKey:MMEResponseKey];
+            NSString *urlString = response.URL.absoluteString;
+            NSNumber *statusCode = @(response.statusCode);
+            NSString *statusCodeString = [statusCode stringValue];
+            NSString *failedRequestKey = [NSString stringWithFormat:@"%@, %@",urlString, statusCodeString];
+            
+            if (self.metrics.failedRequestsDict == nil) {
+                self.metrics.failedRequestsDict = [[NSMutableDictionary alloc] init];
+            }
+        
+            NSNumber *failedCount = [self.metrics.failedRequestsDict objectForKey:failedRequestKey];
+            failedCount = [NSNumber numberWithInteger:[failedCount integerValue] + 1];
+            [self.metrics.failedRequestsDict setObject:failedCount forKey:failedRequestKey];
+        }
+    }
+}
+
+- (void)metricsFromData:(NSData *)data {
+    self.metrics.totalDataTransfer += data.length;
+    
+    if ([[MMEReachability reachabilityForLocalWiFi] isReachableViaWiFi]) {
+        self.metrics.wifiDataTransfer += data.length;
+    } else {
+        self.metrics.cellDataTransfer += data.length;
+    }
+}
+
+- (void)incrementAppWakeUpCount {
+    self.metrics.appWakeups++;
+}
+
+- (void)captureConfigurationJSON:(NSDictionary *)configuration {
+    if (configuration) {
+        self.metrics.configResponseDict = configuration;
+    }
+}
+
+- (void)captureCoordinate:(CLLocationCoordinate2D)coordinate; {
+    if (!self.metrics.deviceLat || !self.metrics.deviceLon) {
+        self.metrics.deviceLat = round(coordinate.latitude*1000)/1000;
+        self.metrics.deviceLon = round(coordinate.longitude*1000)/1000;
+    }
+}
+
+- (void)updateDateUTC {
+    self.metrics.dateUTC = [NSDate date];
```

Comment:
The `dateUTC` property should probably just be called `date` - you use the date formatter to create the UTC string (though it could be any time zone for that matter)

Reply:
Why would the time zone differ? According the the [NSDate docs](https://developer.apple.com/documentation/foundation/nsdate?language=objc) it should always reference UTC timezone until formatted and then it would be changed to the desired timezone.